posthog/hogql_queries/web_analytics/stats_table_pre_aggregated.py

```diff
@@ -75,12 +76,41 @@ def _bounce_rate_query(self) -> str:
 
         return query_str
 
+    def _path_query(self) -> str:
+        previous_period_filter, current_period_filter = self.get_date_ranges(table_name="p")
+
+        query_str = f"""
+        SELECT
+            pathname as `context.columns.breakdown_value`,
+            tuple(
+                uniqMergeIf(p.persons_uniq_state, {current_period_filter}),
+                uniqMergeIf(p.persons_uniq_state, {previous_period_filter})
+            ) AS `context.columns.visitors`,
+            tuple(
+                sumMergeIf(p.pageviews_count_state, {current_period_filter}),
+                sumMergeIf(p.pageviews_count_state, {previous_period_filter})
+            ) as `context.columns.views`,
+            any(bounces.`context.columns.bounce_rate`) as `context.columns.bounce_rate`
+        FROM
+            web_paths_daily p
+        LEFT JOIN ({self._bounce_rate_query()}) bounces
+            ON p.pathname = bounces.`context.columns.breakdown_value`
+        GROUP BY `context.columns.breakdown_value`
+        """
+
+        return query_str
+
     def get_query(self) -> ast.SelectQuery:
         previous_period_filter, current_period_filter = self.get_date_ranges()
 
         query_str = ""
+        table_name = "web_stats_daily"
         if self.runner.query.breakdownBy == WebStatsBreakdown.INITIAL_PAGE:
             query_str = self._bounce_rate_query()
+            table_name = "web_bounces_daily"
+        elif self.runner.query.breakdownBy == WebStatsBreakdown.PAGE:
+            query_str = self._path_query()
+            table_name = "p"
```

Comment:
is this right?

Reply:
Yeah, it's the alias

Reply:
can we make it slightly nicer? I dont think this is very easy to debug :)

Reply:
Yeah, that's the AST comment above. I will refactor this to use the structure and helpers as the stats_table. The other tiles all have simpler queries, that one was the most complex so it is a nice moment to go back and make it cleaner